File: puzzles/kotlin/src/temperatures/temperatures.kt
import java.util.*

fun main(args : Array<String>) {
  val input = Scanner(System.`in`)
  val n = input.nextInt() // the number of temperatures to analyse
  var min = 5526
  for (i in 0 until n) {
    val temp = input.nextInt() // a temperature expressed as an integer ranging from -273 to 5526
    if (Math.abs(temp) < Math.abs(min) || Math.abs(temp) == Math.abs(min) && temp > min) {
      min = temp
    }
  }

  val result = if (n == 0) 0 else min
  println(result)
}

File: puzzles/ts/temperatures/temperatures.ts
const n: number = Number(readline()); // the number of temperatures to analyse
const inputs: string[] = readline().split(' ');

let temperatureClosestToZero: number = 5526;
for (let i = 0; i < n; i++) {
  const temperature: number = Number(inputs[i]);
  const absoluteValue = Math.abs(temperature);
  const closestAbsoluteValue = Math.abs(temperatureClosestToZero);
  if (absoluteValue < closestAbsoluteValue) {
    temperatureClosestToZero = temperature
  } else if (absoluteValue == closestAbsoluteValue && temperature > temperatureClosestToZero) {
    temperatureClosestToZero = temperature;
  }
}

if (n === 0) {
  console.log(0);
} else {
  console.log(temperatureClosestToZero);
}
